includes/Core/Authentication/Google_Proxy.php

```diff
@@ -78,6 +79,67 @@ public function url( $path = '' ) {
 		return $url;
 	}
 
+	/**
+	 * Sends a POST request to the Google Proxy server.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string      $uri Endpoint to send the request to.
+	 * @param Credentials $credentials Credentials instance.
+	 * @param array       $args Array of request arguments.
+	 * @return array|WP_Error The response as an associative array or WP_Error on failure.
+	 */
+	private function request( $uri, $credentials, array $args = array() ) {
+		$request_args = array(
+			'headers' => ! empty( $args['headers'] ) && is_array( $args['headers'] ) ? $args['headers'] : array(),
+			'body'    => ! empty( $args['body'] ) && is_array( $args['body'] ) ? $args['body'] : array(),
+		);
+
+		if ( $credentials && $credentials instanceof Credentials ) {
+			if ( ! $credentials->has() ) {
+				return new WP_Error( 'oauth_credentials_not_exist' );
+			}
+
+			$creds                               = $credentials->get();
+			$request_args['body']['site_id']     = $creds['oauth2_client_id'];
+			$request_args['body']['site_secret'] = $creds['oauth2_client_secret'];
+		}
+
+		if ( ! empty( $args['access_token'] ) ) {
+			$request_args['headers']['Authorization'] = 'Bearer ' . $args['access_token'];
+		}
+
+		if ( isset( $args['mode'] ) && 'async' === $args['mode'] ) {
+			$request_args['timeout']  = 0.01;
+			$request_args['blocking'] = false;
+		}
+
+		if ( ! empty( $args['json_request'] ) ) {
+			$request_args['headers']['Content-Type'] = 'application/json';
+			$request_args['body']                    = wp_json_encode( $request_args['body'] );
+		}
+
+		$url      = $this->url( $uri );
+		$response = wp_remote_post( $url, $request_args );
+		if ( is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		$code = wp_remote_retrieve_response_code( $response );
+		$body = wp_remote_retrieve_body( $response );
+		$body = json_decode( $body, true );
+		if ( $code < 200 || 299 < $code ) {
+			$message = is_array( $body ) && ! empty( $body['error'] ) ? $body['error'] : '';
+			return new WP_Error( 'request_failed', $message, array( 'status' => $code ) );
+		}
+
+		if ( ! $body ) {
```

Comment:
This is throwing an error when the response is empty, eg:
`"body": "{}\n"` which is currently the case for the features endpoint (since no features are active). Wondering if we should make this check more explicitly look for a null? (`$body === null`). cc: @felixarntz

Reply:
@eugene-manuilov See above. Let's maybe change this to a `is_array` check, since the service will always return an JSON array or object of data.

Reply:
@felixarntz @adamsilverstein yep, already updated it to check if it is `is_null( $body )`.